### PR TITLE
Implement stddef.h

### PR DESCRIPTION
**Pull request by robinlinden**

### CODE REVIEW

File: include/stddef.h [hunk 1]
```diff
@@ -0,0 +1,14 @@
+#ifndef _STDDEF_H_
+#define _STDDEF_H_
+
+#undef NULL
+#define NULL 0
+
+#undef offsetof
+#define offsetof(type, member) __builtin_offsetof(type, member)
```
Comment (johnor) [inline include/stddef.h]: Is __builtin_offsetof available on msvc? Looks like __builtin_offsetof is gcc/clang only?

Comment (robinlinden) [reply include/stddef.h]: master doesn't support MSVC yet. :P I'll merge that PR and check if it's avilable. iirc it was added recently.

Comment (robinlinden) [reply include/stddef.h]: Grepping for __builtin_offsetof in their stdlib gave results, but it doesn't seem to exist in VS2017. I used an UB hack instead and it seems to work now.

MSVC was also lacking all the __x_TYPE__ types, so I replaced them too.

Comment (johnor) [reply include/stddef.h]: Cool :)

**Pull request merged**